common/src/main/java/com/wynntils/core/managers/ManagerRegistry.java

```diff
@@ -0,0 +1,172 @@
+    private static final List<Class<? extends Manager>> PERSISTENT_MANAGERS = new ArrayList<>();
+    private static final Map<Class<? extends Manager>, List<Feature>> MANAGER_DEPENDENCIES = new HashMap<>();
+    private static final Set<Class<? extends Manager>> ENABLED_MANAGERS = new HashSet<>();
+
+    public static void init() {
+        registerPersistentDependency(CompassManager.class);
+        registerPersistentDependency(ConfigManager.class);
+        registerPersistentDependency(ClientCommandManager.class);
+        registerPersistentDependency(KeyManager.class);
+        registerPersistentDependency(ModelManager.class);
+        registerPersistentDependency(OverlayManager.class);
+        registerPersistentDependency(WebManager.class);
+
+        addCrashCallbacks();
+    }
+
+    /**
+     * Use this if you want to register a manager that is an essential part of the mod.
+     * <p>
+     * Usually these managers are linked to the internal workings of the mod and not
+     * directly modifying the game itself.
+     * <p>
+     * Do not use this if you don't know what you are doing. Instead, register the manager as a feature dependency.
+     * */
+    private static void registerPersistentDependency(Class<? extends Manager> manager) {
+        PERSISTENT_MANAGERS.add(manager);
+
+        WynntilsMod.getEventBus().register(manager);
+
+        tryInitManager(manager);
+    }
+
+    public static void addDependency(Feature dependant, Class<? extends Manager> dependency) {
+        if (PERSISTENT_MANAGERS.contains(dependency)) {
+            throw new IllegalStateException("Tried to register a persistent manager.");
+        }
+
+        MANAGER_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+
+        MANAGER_DEPENDENCIES.get(dependency).add(dependant);
+
+        updateManagerState(dependency);
+    }
+
+    public static void removeDependency(Feature dependant, Class<? extends Manager> dependency) {
+        if (PERSISTENT_MANAGERS.contains(dependency)) {
+            throw new IllegalStateException("Tried to unregister a persistent manager.");
+        }
+
+        MANAGER_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+
+        MANAGER_DEPENDENCIES.get(dependency).remove(dependant);
+
+        updateManagerState(dependency);
+    }
+
+    public static void removeAllFeatureDependency(Feature dependant) {
+        for (Class<? extends Manager> manager : MANAGER_DEPENDENCIES.keySet()) {
+            boolean removed = MANAGER_DEPENDENCIES.get(manager).remove(dependant);
+
+            if (removed) {
+                updateManagerState(manager);
+            }
+        }
+    }
+
+    private static void updateManagerState(Class<? extends Manager> manager) {
+        List<Feature> dependencies = MANAGER_DEPENDENCIES.get(manager);
+
+        if (ENABLED_MANAGERS.contains(manager)) {
+            if (dependencies == null || dependencies.isEmpty()) {
+                disableManager(manager);
```

Comment:
Are you sure you need to split this into two functions? (same for enable) The difference is not really clear to me.

Reply:
Which two functions are you talking about?

Reply:
Done